File: SignalRDashboard.Data.Demo/Hubs/Models/SiteStatistics.cs
using SignalRDashboard.Data.Core.Hubs.Models;

namespace SignalRDashboard.Data.Demo.Hubs.Models
{
    public class SiteStatistics : ModelBase
    {
        private int _visitors;
        public int Visitors
        {
            get { return _visitors; }
            set { SetProperty(ref _visitors, value); }
        }

        private int _pageViews;
        public int PageViews
        {
            get { return _pageViews; }
            set { SetProperty(ref _pageViews, value); }
        }

        private int _sessions;
        public int Sessions
        {
            get { return _sessions; }
            set { SetProperty(ref _sessions, value); }
        }

        private string _averageSessionDuration;
        public string AverageSessionDuration
        {
            get { return _averageSessionDuration; }
            set { SetProperty(ref _averageSessionDuration, value); }
        }

        private decimal _pagesPerSession;
        public decimal PagesPerSession
        {
            get { return _pagesPerSession; }
            set { SetProperty(ref _pagesPerSession, value); }
        }

        private decimal _bounceRate;
        public decimal BounceRate
        {
            get { return _bounceRate; }
            set { SetProperty(ref _bounceRate, value); }
        }
    }
}

File: SignalRDashboard.Web/Scripts/app/models/demo/SiteStatistics.js
(function(signalrdashboardDemo) {
    signalrdashboardDemo.SiteStatistics = SiteStatistics;

    function SiteStatistics() {
        // Properties matching data model received via SignalR
        this.visitors = 0;
        this.pageViews = 0;
        this.sessions = 0;
        this.averageSessionDuration = "";
        this.pagesPerSession = 0.0;
        this.bounceRate = 0.0;

        this.updateFromData = function(data) {            
            signalrdashboard.mapping.map(data, this);
        };
    }
    
})(window.signalrdashboard.demo || (window.signalrdashboard.demo = {}));